services/streammanager/kafka/kafkamanager.go

```diff
@@ -450,14 +451,17 @@ func serializeAvroMessage(value []byte, codec goavro.Codec) ([]byte, error) {
 	return binary, nil
 }
 
-func prepareBatchOfMessages(topic string, batch []map[string]interface{}, timestamp time.Time, p producerManager) (
+func prepareBatchOfMessages(conf configuration, batch []map[string]interface{}, timestamp time.Time, p producerManager) (
 	[]client.Message, error,
 ) {
 	start := now()
 	defer func() { kafkaStats.prepareBatchTime.SendTiming(since(start)) }()
-
 	var messages []client.Message
+	topic := conf.Topic
 	for i, data := range batch {
+		if conf.MultiTopicSupport {
+			topic = data["topic"].(string)
```

Comment:
shouldn't we fallback to the default topic if the message doesn't contain one?

Reply:
This is handled with transformer being the responsible service for assigning to topic to each message

Reply:
opensource users don't always update their transformers, do they? Plus I don't feel safe with this decision (have commented about it elswhere)